datahub-web-react/src/app/ingest/source/builder/RecipeBuilder.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button, message } from 'antd';
+import React, { useState } from 'react';
+import YAML from 'yamljs';
+import { CodeOutlined, FormOutlined } from '@ant-design/icons';
+import styled from 'styled-components/macro';
+import { ANTD_GRAY } from '../../../entity/shared/constants';
+import { YamlEditor } from './YamlEditor';
+import RecipeForm from './RecipeForm/RecipeForm';
+
+export const ControlsContainer = styled.div`
+    display: flex;
+    justify-content: space-between;
+    margin-top: 8px;
+`;
+
+const BorderedSection = styled.div`
+    display: flex;
+    flex-direction: column;
+    padding-bottom: 16px;
+    border: solid ${ANTD_GRAY[4]} 0.5px;
+`;
+
+const StyledButton = styled(Button)<{ isSelected: boolean }>`
+    ${(props) =>
+        props.isSelected &&
+        `
+        color: #1890ff;
+        &:focus {
+            color: #1890ff;
+        }    
+    `}
+`;
+
+const ButtonsWrapper = styled.div`
+    display: flex;
+    justify-content: flex-end;
+    margin-bottom: 10px;
+`;
+
+interface Props {
+    type: string;
+    isEditing: boolean;
+    displayRecipe: string;
+    setStagedRecipe: (recipe: string) => void;
+    onClickNext: () => void;
+    goToPrevious?: () => void;
+}
+
+function RecipeBuilder(props: Props) {
+    const { type, isEditing, displayRecipe, setStagedRecipe, onClickNext, goToPrevious } = props;
+
+    const [isViewingForm, setIsViewingForm] = useState(true);
+
+    function switchViews(isFormView: boolean) {
+        try {
+            YAML.parse(displayRecipe);
+            setIsViewingForm(isFormView);
+        } catch (e) {
+            message.warn('Found invalid YAML. Please fix your recipe in order to switch views.');
```

Comment:
do we not get an informative error here?

Reply:
it looks like we actually do! I was snagged this from elsewhere since this is what we were doing but we do get the problem line - will update